agent/utils/utils_test.go

```diff
@@ -492,6 +492,41 @@ func TestResponseForwarderWithProxyHangup(t *testing.T) {
 	}
 }
 
+func TestResponseForwarderWithRequestCancellation(t *testing.T) {
+	const (
+		backendID      = "backend"
+		requestID      = "request"
+		endUserMessage = "hello"
+		backendMessage = "ok"
+	)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	endUserRequest := httptest.NewRequest(http.MethodPost, "/", strings.NewReader(endUserMessage)).WithContext(ctx)
+	errsChan := make(chan error, 1)
+	proxyServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer close(errsChan)
+		if _, err := ioutil.ReadAll(r.Body); err != nil {
+			errsChan <- err
+		}
+	}))
+	defer proxyServer.Close()
+	defer proxyServer.CloseClientConnections()
+	proxyClient := proxyServer.Client()
+	_, err := NewResponseForwarder(proxyClient, proxyServer.URL+"/", backendID, requestID, endUserRequest, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cancel()
+	select {
+	case err := <-errsChan:
+		if err != nil {
+			t.Errorf("Unexpected error reading the posted proxy request body: %+v", err)
```

Comment:
Nit: is `%+v` formatting intentional for this error?

Reply:
Yes, because I don't know a type of error that could occur here, and this is being reported as part of marking the test as failing, so I want the message to be as verbose as possible so that anyone seeing it has a better chance of debugging it.